azurerm/internal/services/web/tests/resource_arm_app_service_virtual_network_connection_gateway_test.go

```diff
@@ -0,0 +1,114 @@
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:  func() { acceptance.PreCheck(t) },
+		Providers: acceptance.SupportedProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAzureRMAppServiceVirtualNetworkConnectionGateway_basic(data),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttrSet(data.ResourceName, "name"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "id"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "certificate_blob"),
+					resource.TestCheckResourceAttrSet(data.ResourceName, "certificate_thumbprint"),
+				),
+			},
+			data.ImportStep("virtual_network_gateway_id"),
+		},
+	})
+}
+
+func testAccAzureRMAppServiceVirtualNetworkConnectionGateway_basic(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+resource "azurerm_resource_group" "test" {
+	name     = "acctestRG-appservice-%d"
+	location = "%s"
+}
+
+resource "azurerm_app_service_plan" "test" {
+	name                = "acctestASP-%d"
+	location            = "${azurerm_resource_group.test.location}"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+		
+	sku {
+		tier = "Standard"
+		size = "S1"
+	}
+}
+		
+resource "azurerm_app_service" "test" {
+	name                = "acctestAS-%d"
+	location            = "${azurerm_resource_group.test.location}"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	app_service_plan_id = "${azurerm_app_service_plan.test.id}"
+}
+
+resource "azurerm_virtual_network" "test" {
+	name                = "acctestVnet-%d"
+	resource_group_name = "${azurerm_resource_group.test.name}"
+	location            = "${azurerm_resource_group.test.location}"
+	address_space       = ["10.0.0.0/16"]
+}
+
+resource "azurerm_subnet" "test" {
+  name                 = "GatewaySubnet"
+  resource_group_name  = "${azurerm_resource_group.test.name}"
+  virtual_network_name = "${azurerm_virtual_network.test.name}"
+  address_prefix       = "10.0.1.0/24"
+}
+
+resource "azurerm_public_ip" "test" {
+  name                = "test"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  allocation_method = "Dynamic"
+}
+
+resource "azurerm_virtual_network_gateway" "test" {
+  name                = "test"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+
+  type     = "Vpn"
+  vpn_type = "RouteBased"
+  sku      = "Basic"
+
+  ip_configuration {
+    name                          = "vnetGatewayConfig"
+    public_ip_address_id          = "${azurerm_public_ip.test.id}"
+    private_ip_address_allocation = "Dynamic"
+    subnet_id                     = "${azurerm_subnet.test.id}"
+  }
+
+  vpn_client_configuration {
+    address_space = ["10.2.0.0/24"]
+    vpn_client_protocols = ["SSTP"]
+  }
+
+  lifecycle {
+    ignore_changes = [
+      vpn_client_configuration.0.root_certificate,
```

Comment:
Why are we ignoring lifecycle changes on this property? it might want to be computed if its being updated externally now

Reply:
Because the steps are 
1. CreateOrUpdateVnetConnection
2. result of step 1 contains cert infomation, we should set the cert to virtual network gateway (check duplicate)
3. generate vpn package uri
4. CreateOrUpdateVnetConnectionGateway using step 3's result

a gateway required virtual network integration needs these four steps. In this process, we need update certificate of the virtual network gateway.
I think there is no means to split this process.

I have added `computed: true` in the newest codes for virtual network gateway